code/script.js

```diff
@@ -231,95 +248,169 @@ const setSecret = () => {
-      )
+      message = `No, the person doesn't wear ${value}! Remove all people that wears ${value}.`
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      message = `Yes, the person has ${value}! Keep all people that has ${value}.`
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      message = `No, the person doesn't has ${value}! Remove all people that has ${value}.`
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
+
+  } else {
+    if (keep) { //Hair or eyes
+      message = `Yes the person has ${value} ${category}! Keep all people that has ${value} ${category}.`
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      message = `No, the person doesn't have ${value} ${category}! Remove all people that has ${value} ${category}.`
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
+  showPopUp(message)
+}
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+const showPopUp = (message) => {
+  popUp.style.display = 'block'
+  overlay.style.display = 'block'
+  popUpMessage.innerHTML = message
+}
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+const showPopUpConfirm = (message) => {
+  popUpConfirm.style.display = 'block'
+  overlay.style.display = 'block'
+  confirmMessage.innerHTML = message
+  confirmYesButton.addEventListener('click', checkMyGuess)
+}
 
-  // Invoke a function to redraw the board with the remaining people.
+const closePopUp = () => {
+  popUp.style.display = 'none'
+  overlay.style.display = 'none'
+  popUpConfirm.style.display = 'none'
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+let personToConfirm
```

Comment:
I think that it would be a good thing to position this variable at the beginning with the others, in order to gather all global scope variables at the same place.